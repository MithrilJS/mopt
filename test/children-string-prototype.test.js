"use strict";

var code = require("./lib/code");

describe("Children", () => {
    describe("String.prototype method children", () => {
        it("should support String.prototype methods using dot notation", () =>
            expect(
                code(`m("div", "fooga".replace("f", "g"))`)
            )
            .toMatchSnapshot()
        );
        
        it("should support String.prototype methods using array notation", () =>
            expect(
                code(`m("div", "fooga"["replace"]("f", "g"))`)
            )
            .toMatchSnapshot()
        );
    });
});
