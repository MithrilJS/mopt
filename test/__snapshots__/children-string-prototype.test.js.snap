// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Children String.prototype method children should support String.prototype methods using array notation 1`] = `"m.vnode(\\"div\\",undefined,undefined,undefined,\\"fooga\\"[\\"replace\\"](\\"f\\",\\"g\\"),undefined);"`;

exports[`Children String.prototype method children should support String.prototype methods using dot notation 1`] = `"m.vnode(\\"div\\",undefined,undefined,undefined,\\"fooga\\".replace(\\"f\\",\\"g\\"),undefined);"`;
